### ISSUE TITLE
Error when destroying blocks underneath rail placed at top of world

### ISSUE THREAD

**Issue by AKPWebDesign** [SpongePowered/SpongeCommon#751]
So, if you place blocks at Y=255, then place rail on top of that, when you go to destroy the blocks underneath the rails, the rail is not always destroyed along with it, allowing things like this to happen:
![Image of rails floating in air](https://i.imgur.com/lcENt8I.png)

I'm using [SpongeVanilla@a1ace47e](https://github.com/SpongePowered/SpongeVanilla/tree/a1ace47ecd79943b76678ca6bc52bea0427a941c), which depends on [SpongeCommon@0eccbc74](https://github.com/SpongePowered/SpongeCommon/tree/0eccbc74272f473b07aec1788a3ad072bc3618f0) (I don't believe any changes have been made since then that would affect this). The stack trace from the console when it happens is below. I believe it's a SpongeCommon issue, as I'm only seeing org.spongepowered.common in the stack trace, but let me know if I guessed wrong, and I'll gladly move it to the appropriate issue tracker.

```
[14:41:30] [Server thread/FATAL]: Error executing task
java.util.concurrent.ExecutionException: net.minecraft.util.ReportedException: Exception while updating neighbours
	at java.util.concurrent.FutureTask.report(FutureTask.java:122) ~[?:1.8.0_91]
	at java.util.concurrent.FutureTask.get(FutureTask.java:192) ~[?:1.8.0_91]
	at net.minecraft.util.Util.runTask(Util.java:14) [Util.class:?]
	at net.minecraft.server.MinecraftServer.updateTimeLightAndEntities(MinecraftServer.java:143) [MinecraftServer.class:?]
	at net.minecraft.server.dedicated.DedicatedServer.updateTimeLightAndEntities(DedicatedServer.java:356) [DedicatedServer.class:?]
	at net.minecraft.server.MinecraftServer.tick(MinecraftServer.java:657) [MinecraftServer.class:?]
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:536) [MinecraftServer.class:?]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_91]
Caused by: net.minecraft.util.ReportedException: Exception while updating neighbours
	at org.spongepowered.common.event.CauseTracker.notifyBlockOfStateChange(CauseTracker.java:1127) ~[CauseTracker.class:?]
	at net.minecraft.world.World.notifyNeighborsOfStateChange(World.java:1335) ~[World.class:?]
	at net.minecraft.world.World.notifyNeighborsRespectDebug(World.java:416) ~[World.class:?]
	at net.minecraft.world.World.markAndNotifyNeighbors(World.java:1198) ~[World.class:?]
	at org.spongepowered.common.event.CauseTracker.markAndNotifyBlockPost(CauseTracker.java:1180) ~[CauseTracker.class:?]
	at org.spongepowered.common.event.CauseTracker.handleBlockCaptures(CauseTracker.java:870) ~[CauseTracker.class:?]
	at org.spongepowered.common.event.CauseTracker.handlePostTickCaptures(CauseTracker.java:404) ~[CauseTracker.class:?]
	at org.spongepowered.common.network.PacketUtil.onProcessPacket(PacketUtil.java:116) ~[PacketUtil.class:?]
	at net.minecraft.network.PacketThreadUtil$1.redirect$onProcessPacket$0(PacketThreadUtil.java:39) ~[PacketThreadUtil$1.class:?]
	at net.minecraft.network.PacketThreadUtil$1.run(PacketThreadUtil.java:15) ~[PacketThreadUtil$1.class:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[?:1.8.0_91]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[?:1.8.0_91]
	at net.minecraft.util.Util.runTask(Util.java:13) ~[Util.class:?]
	... 5 more
Caused by: org.spongepowered.api.util.PositionOutOfBoundsException: Position is out of bounds: expected in range (-30000000, 0, -30000000) to (29999999, 255, 29999999) but got (161, 256, 256)
	at net.minecraft.world.World.checkBlockBounds(World.java:685) ~[World.class:?]
	at net.minecraft.world.World.getBlock(World.java:354) ~[World.class:?]
	at org.spongepowered.api.world.extent.BlockVolume.getBlock(BlockVolume.java:96) ~[BlockVolume.class:?]
	at org.spongepowered.api.world.Location.getBlock(Location.java:451) ~[Location.class:?]
	at org.spongepowered.common.block.SpongeBlockSnapshotBuilder.from(SpongeBlockSnapshotBuilder.java:114) ~[SpongeBlockSnapshotBuilder.class:?]
	at org.spongepowered.common.block.SpongeBlockSnapshotBuilder.from(SpongeBlockSnapshotBuilder.java:63) ~[SpongeBlockSnapshotBuilder.class:?]
	at net.minecraft.block.Block.redirect$onSpawnAsEntity$0(Block.java:213) ~[Block.class:?]
	at net.minecraft.block.Block.spawnAsEntity(Block.java:597) ~[Block.class:?]
	at net.minecraft.block.Block.dropBlockAsItemWithChance(Block.java:577) ~[Block.class:?]
	at net.minecraft.block.Block.dropBlockAsItem(Block.java:557) ~[Block.class:?]
	at net.minecraft.block.BlockRailBase.onNeighborBlockChange(BlockRailBase.java:136) ~[BlockRailBase.class:?]
	at org.spongepowered.common.event.CauseTracker.notifyBlockOfStateChange(CauseTracker.java:1111) ~[CauseTracker.class:?]
	at net.minecraft.world.World.notifyNeighborsOfStateChange(World.java:1335) ~[World.class:?]
	at net.minecraft.world.World.notifyNeighborsRespectDebug(World.java:416) ~[World.class:?]
	at net.minecraft.world.World.markAndNotifyNeighbors(World.java:1198) ~[World.class:?]
	at org.spongepowered.common.event.CauseTracker.markAndNotifyBlockPost(CauseTracker.java:1180) ~[CauseTracker.class:?]
	at org.spongepowered.common.event.CauseTracker.handleBlockCaptures(CauseTracker.java:870) ~[CauseTracker.class:?]
	at org.spongepowered.common.event.CauseTracker.handlePostTickCaptures(CauseTracker.java:404) ~[CauseTracker.class:?]
	at org.spongepowered.common.network.PacketUtil.onProcessPacket(PacketUtil.java:116) ~[PacketUtil.class:?]
	at net.minecraft.network.PacketThreadUtil$1.redirect$onProcessPacket$0(PacketThreadUtil.java:39) ~[PacketThreadUtil$1.class:?]
	at net.minecraft.network.PacketThreadUtil$1.run(PacketThreadUtil.java:15) ~[PacketThreadUtil$1.class:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[?:1.8.0_91]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[?:1.8.0_91]
	at net.minecraft.util.Util.runTask(Util.java:13) ~[Util.class:?]
	... 5 more
```

**Comment by ryantheleach** [comment]
I remember strange things happening on craftbukkit near the build height in regards to blocks that attach to the blocks near the ceiling so I investigated how vanilla currently behaves.

Tested on 1.8.9 single player.

Placing rail at 256 fails with the message Height limit for building is 256 blocks.

![image](https://cloud.githubusercontent.com/assets/141554/15663274/72f9af56-2738-11e6-8114-a8b501e769f0.png)
